assemble/bin/accumulo

```diff
@@ -15,173 +15,567 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Start: Resolve Script Directory
-SOURCE="${BASH_SOURCE[0]}"
-while [ -h "${SOURCE}" ]; do # resolve $SOURCE until the file is no longer a symlink
-   bin="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"
-   SOURCE="$(readlink "${SOURCE}")"
-   [[ "${SOURCE}" != /* ]] && SOURCE="${bin}/${SOURCE}" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
-done
-bin="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"
-script=$( basename "${SOURCE}" )
-# Stop: Resolve Script Directory
+function build_native() {
+  native_tarball="$basedir/lib/accumulo-native.tar.gz"
+  final_native_target="$basedir/lib/native"
 
-. "${bin}"/config.sh
+  if [[ ! -f $native_tarball ]]; then
+      echo "Could not find native code artifact: ${native_tarball}";
+      exit 1
+  fi
 
-START_JAR="${ACCUMULO_HOME}/lib/accumulo-start.jar"
+  # Make the destination for the native library
+  mkdir -p "${final_native_target}" || exit 1
 
-#
-# Resolve a program to its installation directory
-#
-locationByProgram()
-{
-   RESULT=$( which "$1" )
-   if [[ "$?" != 0 && -z "${RESULT}" ]]; then
-      echo "Cannot find '$1' and '$2' is not set in $ACCUMULO_CONF_DIR/accumulo-env.sh"
+  # Make a directory for us to unpack the native source into
+  TMP_DIR=$(mktemp -d /tmp/accumulo-native.XXXX) || exit 1
```

Comment:
Try adding `trap "rm -rf '""$TMP_DIR""'" 0 1 2 3 15` after this, so that the temp directory is cleaned up even if the script exits abnormally.
See https://www.tutorialspoint.com/unix/unix-signals-traps.htm for which signals those correspond to.